core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/config/AwsClientOption.java

```diff
@@ -21,16 +21,29 @@
 import software.amazon.awssdk.awscore.client.builder.AwsClientBuilder;
 import software.amazon.awssdk.awscore.defaultsmode.DefaultsMode;
 import software.amazon.awssdk.core.client.config.ClientOption;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+import software.amazon.awssdk.identity.spi.IdentityProvider;
 import software.amazon.awssdk.regions.Region;
 
 @SdkProtectedApi
 public final class AwsClientOption<T> extends ClientOption<T> {
+    // TODO: Should the existing option be removed?
+    //       Or replaced with same name (CREDENTIALS_PROVIDER) but new type below?
+    //       This class is SdkProtectedApi, and it seems customer cannot create an option with this directly (we do it in
+    //       AwsDefaultClientBuilder). Is there risk of old generated code when there are mixed versions of modules?
+    // If it is kept, should it be marked @Deprecated?
     /**
      * @see AwsClientBuilder#credentialsProvider(AwsCredentialsProvider)
      */
     public static final AwsClientOption<AwsCredentialsProvider> CREDENTIALS_PROVIDER =
             new AwsClientOption<>(AwsCredentialsProvider.class);
```

Comment:
Discussed this one a bit above. +1 to marking it as deprecated, but like I mentioned I think we need to keep it around and keep setting it. We also need to keep reading it, in case customers are using old client modules with the updated aws-core module.